// basic-react-timepicker/lib/TimePicker.js
'use strict';

Object.defineProperty(exports, '__esModule', {
	value: true
});
var React = require('react');
var moment = require('moment');

var TimePicker = React.createClass({
	displayName: 'TimePicker',

	propTypes: {
		defaultValue: React.PropTypes.string,
		onChange: React.PropTypes.func,
		name: React.PropTypes.string,
		beginLimit: React.PropTypes.string,
		endLimit: React.PropTypes.string,
		step: React.PropTypes.number
	},

	isEarlierThanEndLimit: function isEarlierThanEndLimit(timeValue, endLimit, lastValue) {
		var timeValueIsEarlier = moment(timeValue, 'h:mmA').diff(moment(endLimit, 'h:mmA')) < 0;
		var timeValueIsLaterThanLastValue = lastValue === undefined ? true : moment(lastValue, 'h:mmA').diff(moment(timeValue, 'h:mmA')) < 0;
		return timeValueIsEarlier && timeValueIsLaterThanLastValue;
	},

	render: function render() {
		var timeValue = this.props.beginLimit || "12:00AM";
		var lastValue;
		var endLimit = this.props.endLimit || "11:59PM";
		var step = this.props.step || 15;

		var options = [];
		options.push(React.createElement(
			'option',
			{ key: timeValue, value: timeValue },
			timeValue
		));

		while (this.isEarlierThanEndLimit(timeValue, endLimit, lastValue)) {
			lastValue = timeValue;
			console.log(timeValue, moment(timeValue, 'h:mmA').diff(moment(endLimit, 'h:mmA'), 'minutes'));
			timeValue = moment(timeValue, 'h:mmA').add(step, 'minutes').format('h:mmA');
			options.push(React.createElement(
				'option',
				{ key: timeValue, value: timeValue },
				timeValue
			));
		}
		return React.createElement(
			'select',
			{ defaultValue: this.props.defaultValue, onChange: this.props.onChange, name: this.props.name },
			options
		);
	}
});

exports['default'] = TimePicker;
module.exports = exports['default'];
